hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/ObjectStore.java

```diff
@@ -668,19 +668,23 @@ private List<OzoneSnapshot> getNextListOfSnapshots(String prevSnapshot)
    * @param token to get the index to return diff report from.
    * @param pageSize maximum entries returned to the report.
    * @param forceFullDiff request to force full diff, skipping DAG optimization
+   * @param forceNonNativeDiff request to force diff to perform diffs without
+   *                           native lib
    * @return the difference report between two snapshots
    * @throws IOException in case of any exception while generating snapshot diff
    */
+  @SuppressWarnings("parameternumber")
   public SnapshotDiffResponse snapshotDiff(String volumeName,
                                            String bucketName,
                                            String fromSnapshot,
                                            String toSnapshot,
                                            String token,
                                            int pageSize,
-                                           boolean forceFullDiff)
+                                           boolean forceFullDiff,
+                                           boolean forceNonNativeDiff)
```

Comment:
Note adding this param without having a shim to keep compatiblity WILL break applications that are already using `snapshotDiff()`.

```suggestion
                                           boolean diffDisableNativeLibs)
```

Reply:
done